src/proxy/libp2p_transport_proxy.erl

```diff
@@ -41,22 +41,23 @@ connect(_Pid, MAddr, _Options, _Timeout, TID) ->
     Swarm = libp2p_swarm:swarm(TID),
     ID = crypto:strong_rand_bytes(16),
     Args = [
-        {p2p_circuit, MAddr}
-        ,{transport, self()}
-        ,{id, ID}
+        {p2p_circuit, MAddr},
+        {transport, self()},
+        {id, ID}
     ],
-    {ok, _} = libp2p_proxy:dial_framed_stream(
-        Swarm
-        ,PAddress
-        ,Args
-    ),
-    receive
-        {proxy_negotiated, Socket, MultiAddr} ->
-            Conn = libp2p_transport_tcp:new_connection(Socket, MultiAddr),
-            lager:info("proxy successful ~p", [Conn]),
-            libp2p_transport:start_client_session(TID, MAddr, Conn)
-    after 8000 ->
-        {error, timeout_relay_session}
+    case libp2p_proxy:dial_framed_stream(Swarm, PAddress, Args) of
+        {error, Reason} ->
+            lager:error("failed to dial proxy server ~p ~p", [PAddress, Reason]),
+            {error, fail_dial_proxy};
+        {ok, _} ->
+            receive
+                {proxy_negotiated, Socket, MultiAddr} ->
+                    Conn = libp2p_transport_tcp:new_connection(Socket, MultiAddr),
+                    lager:info("proxy successful ~p", [Conn]),
+                    libp2p_transport:start_client_session(TID, MAddr, Conn)
+            after 8000 ->
```

Comment:
do we want this hardcoded?

Reply:
yeah that doesn't seem right?

Reply:
Hmmm.. on second reading I think it actually is correct.. proxying is a tcp specific feature. @macpie 
what happens if that multiaddr isn't a /ip4 multiaddr? Or is that guaranteed at this point?

Reply:
it should be guaranteed at that point